server/chat-plugins/room-events.ts

```diff
@@ -88,25 +136,26 @@ export const commands: ChatCommands = {
 			}
 			if (!target) return this.errorReply("Usage: /roomevents start [event name]");
 			target = toID(target);
-			if (!room.events[target]) return this.errorReply(`There is no such event named '${target}'. Check spelling?`);
-			if (room.events[target].started) {
-				return this.errorReply(`The event ${room.events[target].eventName} has already started.`);
+			const event = room.events[getEventIDByNameOrAlias(target, room)];
+			if (!event) return this.errorReply(`There is no such event named '${target}'. Check spelling?`);
+			if (event.started) {
+				return this.errorReply(`The event ${event.eventName} has already started.`);
 			}
 			for (const u in room.users) {
 				const activeUser = Users.get(u);
 				if (activeUser?.connected) {
 					activeUser.sendTo(
 						room,
 						Chat.html`|notify|A new roomevent in ${room.title} has started!|` +
-						`The "${room.events[target].eventName}" roomevent has started!`
+						`The "${event.eventName}" roomevent has started!`
 					);
 				}
 			}
 			this.add(
-				Chat.html`|raw|<div class="broadcast-blue"><b>The "${room.events[target].eventName}" roomevent has started!</b></div>`
+				Chat.html`|raw|<div class="broadcast-blue"><b>The "${event.eventName}" roomevent has started!</b></div>`
 			);
-			this.modlog('ROOMEVENT', null, `started "${target}"`);
-			room.events[target].started = true;
+			this.modlog('ROOMEVENT', null, `started "${toID(event.eventName)}"`);
```

Comment:
I don't think this needs to be changed to an ID, can I ask your reasoning? I might be missing something.

Reply:
the only reason I made it an ID is because the original code used an ID there

Reply:
oh ok